src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/IActionResultExecutor.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Mvc.Infrastructure
+{
+    public interface IActionResultExecutor<TResult> where TResult : IActionResult
```

Comment:
Is there any value in making this contravariant? Maybe for scenarios like `IActionResultExecutor<NotFoundResult> = new StatusCodeResultExecutor();`

Reply:
Hmm, I don't think it can hurt.